### PR TITLE
URL parameter support

### LINKED ISSUE #29: URL parameter parsing
URL variables should be parsed out of the request and an interface for accessing them should be exposed

### PR DESCRIPTION
**Pull request by despeset**
Closes #29

### CODE REVIEW

File: RequestParser.jl [hunk 1]
```diff
@@ -106,7 +106,27 @@ on_body_cb = cfunction(on_body, HTTP_DATA_CB...)
 function on_message_complete(parser)
     r = partials[parser]
     delete!(r.headers, "current_header", nothing)
+
+    # Handle URL variables eg. `foo/bar?a=b&c=d`
+    m = match(r"\?.*=.*", r.resource)
+    url_params = (String => Union(String,Array))[]
```
Comment (zachallaun) [inline RequestParser.jl]: `Union(String, Array{String})` ?

File: RequestParser.jl [hunk 2]
```diff
@@ -106,7 +106,27 @@ on_body_cb = cfunction(on_body, HTTP_DATA_CB...)
...
+    if m != nothing
+        for set in split(split(r.resource, "?")[2], "&")
+            set = split(set, "=")
```
Comment (zachallaun) [inline RequestParser.jl]: `key, value = split(set, "=")`

File: RequestParser.jl [hunk 3]
```diff
@@ -106,7 +106,27 @@ on_body_cb = cfunction(on_body, HTTP_DATA_CB...)
...
+            # Promote multiple instances of the same key to Array
+            if get(url_params, set[1], nothing) == nothing
```
Comment (zachallaun) [inline RequestParser.jl]: `!has(url_params, key)`

**Comment by zachallaun** [comment]
This looks good to me now.

**Comment by despeset** [comment]
Okay, took out the multiple key support because _some people_ ( cough cough ) think developers shouldn't have to type check every time they use a url_param.  Sheesh!  I added `Request.state[:raw_resource]` so enterprising web apps can still pull multiple values themselves.

**Pull request merged**